docs/wiki/CL/merkleization.md

```diff
@@ -0,0 +1,440 @@
+# Merkleization and Hash Tree Roots 
+
+In Ethereum consensus mechanism, it's critical for all participating nodes to agree on the state of the system consistently and efficiently. The [Simple Serialize (SSZ)](/docs/wiki/CL/SSZ.md) framework facilitates this through Merkleization, a process that transforms serialized data into a Merkle tree structure. This wiki page discusses the intricacies of Merkleization and its importance in ensuring a shared state across nodes in a scalable and secure manner.
+
+## Terminology and Methods
+
+- **Merkleization:** Refers to constructing a Merkle tree and deriving its root.
+- **Hash Tree Root:** A specific application of Merkleization, used to compute the root hash of complex SSZ container.
+
+## The Need for Merkleization
+
+Cryptographic hash functions provide a solution by generating a compact, unique representation of a data set for a Beacon state. By hashing the serialized state of a Beacon chain, nodes can quickly and efficiently compare states by exchanging these small hash outputs.
+
+## Process of Merkleization
+
+Merkleization involves breaking down the serialized data into 32-byte chunks, which serve as the leaves of a Merkle tree. These chunks are then combined pair-wise, hashed together, and the process is repeated up the tree until a single hash—the Merkle root—is derived. This root hash acts as a unique fingerprint for the entire dataset. The key steps are as below:
+
+- **Chunking:** Divide the serialized data into 32-byte chunks.
+- **Tree Construction:** Pair up the chunks and hash each pair to form the next level of the tree. Repeat this step until only one hash remains: the Merkle root.
+- **Padding:** If the number of chunks isn't a power of two, additional zero-value chunks are added to round out the tree, ensuring that the tree is balanced.
+
+## Benefits of Merkleization
+
+- **Performance Efficiency:** While the tree requires hashing approximately twice the original data amount, caching mechanisms can store the roots of subtrees that don't change often. This significantly reduces the computational overhead as only altered parts of the data need re-hashing.
+- **Light Client Support:** The Merkle tree structure supports the creation of Merkle proofs—small pieces of data that prove the inclusion and integrity of specific parts of the state without needing the entire dataset. This feature is crucial for light clients, which operate with limited resources and rely on these proofs to interact with  Ethereum securely.
+
+## Merkle Tree Structure and Hashing
```

Comment:
I am not sure if we need to cover merkle tree basics here. It's also mentioned on EL/data-structures and maybe you can just reference an external resource to keep the article shorter. 
We wrote a merkle tree primer for Summer of protocol, I will link it when it's published